webcompat/form.py

```diff
@@ -154,6 +159,22 @@ def normalize_url(url):
     return url
 
 
+def normalize_metadata(metadata_value):
+    """Normalize the metadata received from the form."""
+    # Removing closing comments.
```

Comment:
Why do we remove just the closing comments? Just wondering about the context here.

Reply:
@zoepage good question.
The reason is that an opening comment will not do much as just being hidden by the other opening comment, but a closing comment will terminate the opening comment.

This is what our issues look like.

```
<!-- @browser: Firefox 59.0 -->
<!-- @ua_header: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:59.0) Gecko/20100101 Firefox/59.0 -->
<!-- @reported_with: desktop-reporter -->
<!-- @extra_label: type-stylo -->

**URL**: https://web.telegram.org/

**Browser / Version**: Firefox 59.0
**Operating System**: Windows 10
**Tested Another Browser**: Yes

**Problem type**: Site is not usable
**Description**: Can't type text, can't delete.
**Steps to Reproduce**:
layout.css.servo.enabled: true



_From [webcompat.com](https://webcompat.com/) with ❤️_
```

And also note that if someone still try to put slide in an opening `<!--` comment. It will be caught by the `<` which will remove everything.

Maybe we need a better strategy, like a dropdown controlled list, though it will limit people. Freedom and expressiveness or Control and security. Life 🗺

Reply:
> The reason is that an opening comment will not do much as just being hidden by the other opening comment, but a closing comment will terminate the opening comment.

But wouldn't the missing closing comment mean, we never close, so everything that follows is also a comment? So nothing would actually gets displayed?

> caught by the < which will remove everything.

Everything = the value of the input  as a param? So `< Firefox 59.0` -> ``?

Reply:
In the body up there

```
<!-- @browser: Firefox 59.0 -->
```

the part we sanitized is `Firefox 59.0` or if you prefer the `string_here`:

```
<!-- @browser: string_here -->
```

The `<!-- @browser: ` and ` -->` are coming from 

https://github.com/webcompat/webcompat.com/blob/a18d81afc2c75f620ba64d4c8b50228fe622e2b8/webcompat/form.py#L117-L122

So let's say we got 
```python
string_here = '--> blah blah'
```
then the resulting string will be 

```
<!-- @browser: --> blah blah -->
```

which will be rendered `blah blah -->` with a hidden useless `<!-- @browser: -->`

Now if 

```python
# opening comment
string_here = 'blah <!-- foo'
```

it will become

```
<!-- @browser: blah <!-- foo -->
```
which does nothing apart of staying hidden.

Try this in your URL bar and do view source.

```
data:text/html,<!-- @browser: blah <!-- foo -->
```

Reply:
Ah! That makes sense. Thanks for the explanation. :)

I'd +1 the PR && say we chat in Austin about ways to improve?